### PR TITLE
DEV-290 Override mint and add enumerable

### PR DESCRIPTION
**Pull request by grp06**

### CODE REVIEW

File: src/Raft.sol [hunk 1]
```diff
@@ -39,6 +41,15 @@ contract Raft is ERC721URIStorage, Ownable, Pausable {
     return newItemId;
   }
 
+  function _setTokenURI(uint256 tokenId, string memory _tokenURI) internal virtual {
+    require(_exists(tokenId), "ERC721URIStorage: URI set of nonexistent token");
```
Comment (rahulrumalla) [inline src/Raft.sol:45]:
```suggestion
    require(_exists(tokenId), "_setTokenURI: URI set of nonexistent token");
```

File: src/Badges.sol [hunk 2]
```diff
@@ -41,30 +41,19 @@ contract Badges is ERC4973, Ownable {
...
-    uint256 raftTokenId = _specToRaft[specUri];
+  function _mint(
+    address to,
+    uint256 tokenId,
+    string memory uri
+  ) internal override returns (uint256) {
+    uint256 raftTokenId = _specToRaft[uri];
 
     // only registered specs can be used for minting
     require(raftTokenId != 0, "mintAuthorizedBadge: spec is not registered");
```
Comment (rahulrumalla) [inline src/Badges.sol:52]:
```suggestion
    require(raftTokenId != 0, "_mint: spec is not registered");
```

File: test/badges.test.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ const errMintFailed = 'mintAuthorizedBadge: badge minting failed'
 const errSpecNotRegistered = 'mintAuthorizedBadge: spec is not registered'
 const errSpecAlreadyRegistered = 'createSpecAsRaftOwner: spec already registered'
 const errNotRaftOwner = 'createSpecAsRaftOwner: unauthorized'
+const errInvalidSig = '_safeCheckAgreement: invalid signature'
```
Comment (rahulrumalla) [inline test/badges.test.ts:21]: `errMintFailed` and `errSpecNotRegistered`

I think we can remove these 2 as they'd be unused right?

**Review by rahulrumalla — CHANGES_REQUESTED** [review]
Missing changelog entry as well for version bump.

File: test/badges.test.ts (hunk 3 above)
Comment (grp06) [reply test/badges.test.ts:21]: we use `errSpecNotRegistered` in the final test

Comment (rahulrumalla) [reply test/badges.test.ts:21]: ah right. That needs to be get updated.. the prefix needs to be `_mint`

File: readme.md [hunk 4]
```diff
@@ -42,10 +43,18 @@ npm i @otterspace-xyz/contracts
...
+Badges ABI
+
 ```js
 import Badges from '@otterspace-xyz/contracts/out/Badges.sol/Badges.json' assert { type: 'json' }
 ```
 
+Raft ABI
+
+```js
+import Badges from '@otterspace-xyz/contracts/out/Raft.sol/Raft.json' assert { type: 'json' }
```
Comment (rahulrumalla) [inline readme.md:55]:
```suggestion
import Raft from '@otterspace-xyz/contracts/out/Raft.sol/Raft.json' assert { type: 'json' }
```

**Review by rahulrumalla — APPROVED** [review]
LGTM

**Pull request merged**